crates/services/p2p/src/service.rs

```diff
@@ -444,7 +450,7 @@ impl<V, T> UninitializedTask<V, SharedState, T> {
     }
 }
 
-impl<P: TaskP2PService, V, B: Broadcast, T> Task<P, V, B, T> {
+impl<P: TaskP2PService, V: AtomicView, B: Broadcast, T> Task<P, V, B, T> {
```

Comment:
[nit]: Would be better to move all constraints to `where`

Reply:
addressed in b677c01